src/components/helpers/useClickOutside.ts

```diff
@@ -0,0 +1,22 @@
+import { RefObject, useEffect } from "react";
+
+interface ClickOutsideProps {
+  refEl: RefObject<HTMLDivElement>;
+  callback: () => void;
+}
+
+export const useClickOutside = ({ refEl, callback }: ClickOutsideProps) => {
+  useEffect(() => {
+    const handleClickOutside = (event: MouseEvent) => {
```

Comment:
Tu typ raczej powinien być reactowy: React.MouseEvent<HTMLElement>

wtedy też to poniższe rzutowanie nie będzie potrzebne